massa-async-pool/src/pool.rs

```diff
@@ -123,13 +133,13 @@ impl AsyncPool {
         // gather all selected items and remove them from self.messages
         // iterate in decreasing priority order
         self.messages
-            .drain_filter(|_, msg| {
+            .drain_filter(|_, message| {
```

Comment:
Drain filter remove elements from `self.messages` maybe we should XOR their hash also ?

Reply:
take_batch_to_execute is used only speculatively, hash computations are done on finality